### PR TITLE
mlxbf-bootimages: init at 4.0.3-12704

### PR DESCRIPTION
**Pull request by nikstur**
###### Description of changes

Add mlxbf-bootimages which are required to build Nvidia BlueField bootstreams. Because licensing is unclear, they are marked as unfree for now.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux (via binfmt)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/mlxbf-bootimages/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{ stdenv
+, lib
+, fetchurl
+, dpkg
+}:
+
+let
+  mainVersion = "4.0.2";
```
Comment (RaitoBezarius) [inline pkgs/misc/mlxbf-bootimages/default.nix:8]: I would rather put mainVersion in the version and compute mainVersion from version so you can override properly.

Comment (nikstur) [reply pkgs/misc/mlxbf-bootimages/default.nix]: Good point.

File: pkgs/misc/mlxbf-bootimages/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  '';
+
+  meta = with lib; {
+    description = "BlueField boot images";
+    homepage = "https://github.com/Mellanox/bootimages";
+    # It is unclear if the bootimages themselves are Open Source software. They
+    # never explicitly say they are. They contain Open Source software licensed
+    # under bsd2, bsd2Patent, bsd3. However, it is probably safer to assume
+    # they are unfree.
+    license = licenses.unfree;
```
[the remark this answers is not in the document]
Comment (nikstur) [reply pkgs/misc/mlxbf-bootimages/default.nix:36]: Done

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 233387` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>mlxbf-bootimages</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
You fucked up your download URL

**Comment by nikstur** [comment]
> You fucked up your download URL

Works locally for me. Could this have failed because of either the platform or because its unfree?

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 233387` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>mlxbf-bootimages</li>
  </ul>
</details>

**Comment by nikstur** [comment]
After observing, the fun thing is that not only did I mess up the URL but they also yanked the 4.0.2 release. So now we're at 4.0.3.

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 233387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by RaitoBezarius** [comment]
> After observing, the fun thing is that not only did I mess up the URL but they also yanked the 4.0.2 release. So now we're at 4.0.3.

The power of content addressability

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**